homeassistant/components/homekit/util.py

```diff
@@ -414,7 +414,11 @@ def cleanup_name_for_homekit(name: str | None) -> str:
     # likely isn't a problem
     if name is None:
         return "None"  # None crashes apple watches
-    return name.translate(HOMEKIT_CHAR_TRANSLATIONS)[:MAX_NAME_LENGTH]
+    return (
+        name.translate(HOMEKIT_CHAR_TRANSLATIONS)
+        .lstrip("-_")[:MAX_NAME_LENGTH]
```

Comment:
We should strip before we truncate

Reply:
`rstrip` is kept after truncating so that the name doesn't end up with a `-` or `_` in the truncated string.